front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/PipelineController.groovy

```diff
@@ -44,6 +44,18 @@ class PipelineController {
     pipelineDAO.getPipelinesByApplication(application)
   }
 
+  @RequestMapping(value = 'getById/{id:.+}', method = RequestMethod.GET)
```

Comment:
ok. I need to create a new PR unless you know how I can checkout a branch with a slash. I didnt realize I'd hit this problem when I created the branch. Stackoverflow suggests it should be working anyway, but git seems to be interpreting the branch as a path. I'm going to give up and create a new PR.

Why are you using {application}/{id}? Arent ID's global?

There is already DELETE /pipelines/{application}/{pipeline}. Wouldnt .../{id} be redundant to that?
I thought I saw somewhere that tried to treat the value as an id and if that fails to treat it as a name. I dont recall where I saw that. I didnt like that, but if you are looking to transition from using names to using ids then that would be one way to do it. Since the IDs are internal and the names are external, I think having interfaces by name (at least variants) makes sense.